test/runner.mjs

```diff
@@ -0,0 +1,36 @@
+import { deepStrictEqual as eq } from 'assert'
+import { basename } from 'path'
+
+const pathLength = import.meta.url.length - basename(import.meta.url).length
+
+const record = {}
+export const test = (description, fn, expect) => {
+  const fail = new Error(description)
+  const [, file, l, c] = fail.stack
+    .split('\n')[2]
+    .trim()
+    .slice(pathLength + 3)
+    .split(/(^[^:]+)\:(\d+)\:(\d+)$/)
+
+  const tests = record[file] || (record[file] = [])
```

Comment:
I use the stack to find out what files the tests comes from, then store tests by files.

For now we only have one file, but once we start having more files, it will help to keep it clean